PhysicsTools/PatAlgos/python/slimming/puppiForMET_cff.py

```diff
@@ -39,8 +40,19 @@ def makePuppiesFromMiniAOD( process ):
     process.puppiMerged = cms.EDProducer("CandViewMerger",src = cms.VInputTag( 'puppiNoLep','pfLeptonsPUPPET'))
     process.load('CommonTools.PileupAlgos.PhotonPuppi_cff')
     process.puppiForMET = process.puppiPhoton.clone()
+    process.puppiForMET.candName = cms.InputTag('packedPFCandidates')
+    process.puppiForMET.photonName = cms.InputTag('slimmedPhotons')
+    process.puppiForMET.runOnMiniAOD = cms.bool(True)
     setupPuppiPhotonMiniAOD(process)
-    #Line below doesn't work because of an issue with references in MiniAOD without setting useRefs=>False and using delta R
-    #process.puppiForMET.puppiCandName    = 'puppiMerged'
-    #process.puppiForMET.useRefs          = False
+    #Line below replaces reference linking wiht delta R matching because the puppi references after merging are not consistent with those of the original packed candidate collection
+    process.puppiForMET.useRefs          = False
+    #Line below points puppi MET to puppi no lepton which increases the response
+    process.puppiForMET.puppiCandName    = 'puppiMerged'
+    #Avoid recomputing the weights available in MiniAOD
+    #process.puppi.useExistingWeights = True
+    #process.puppiNoLep.useExistingWeights = True
```

Comment:
This is related to the puppi tuning that is not the same between the RECO step and the miniAOD step. As the backport now contains the proper tuning, I turned on those lines